src/Workspaces/Core/MSBuild.BuildHost/Build/ProjectBuildManager.cs

```diff
@@ -103,7 +103,22 @@ public ProjectBuildManager(ImmutableDictionary<string, string> additionalGlobalP
                 // path explicitly is necessary so that the reserved properties like $(MSBuildProjectDirectory) will work.
                 xml.FullPath = path;
 
-                var project = new MSB.Evaluation.Project(xml, globalProperties: null, toolsVersion: null, projectCollection);
+                // Roughly matches the VS project load settings for their design time builds.
+                var projectLoadSettings = MSB.Evaluation.ProjectLoadSettings.RecordDuplicateButNotCircularImports
+                    | MSB.Evaluation.ProjectLoadSettings.RejectCircularImports
+                    | MSB.Evaluation.ProjectLoadSettings.IgnoreEmptyImports
+                    | MSB.Evaluation.ProjectLoadSettings.IgnoreMissingImports
+                    | MSB.Evaluation.ProjectLoadSettings.IgnoreInvalidImports
+                    | MSB.Evaluation.ProjectLoadSettings.RecordEvaluatedItemElements
```

Comment:
So many of these seem like error handling flags, but what does this record? Ditto for RecordDuplicateButNotCircularImports? If that's recording data that we'd have to do something special to read then maybe we don't need it. But if it just reports to the binlog, great.

Reply:
Ah we don't need the record values.  Looks like it just updates a property on the project which we do not read.
https://github.com/dotnet/msbuild/blob/221fd2e8790a22ead513eb71630557efc02060e2/src/Build/Evaluation/Evaluator.cs#L2208C39-L2208C65

Same for the RecordEvaluatedItemElements